html/modules/custom/reliefweb_job_tagger/reliefweb_job_tagger.module

```diff
@@ -0,0 +1,115 @@
+<?php
+
+/**
+ * @file
+ * OCHA AI Job tagging.
+ */
+
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\reliefweb_moderation\Helpers\UserPostingRightsHelper;
+use Drupal\reliefweb_utility\Helpers\MarkdownHelper;
+use Drupal\reliefweb_utility\Helpers\UserHelper;
+
+/**
+ * Implements hook_form_HOOK_alter().
+ */
+function reliefweb_job_tagger_form_node_form_alter(array &$form, FormStateInterface $form_state, $form_id) {
+  $form_ids = [
+    'node_job_edit_form',
+    'node_job_form',
+  ];
+
+  if (!in_array($form_id, $form_ids)) {
+    return;
+  }
+
+  $user = \Drupal::currentUser();
+
+  if ($form_id == 'node_job_edit_form') {
+    $entity = $form_state->getFormObject()->getEntity();
+    if ($entity->hasField('reliefweb_job_tagger_info') && !$entity->reliefweb_job_tagger_info->IsEmpty()) {
+      if (UserHelper::userHasRoles(['editor'])) {
+        $message = $entity->reliefweb_job_tagger_info->value;
+        $message = MarkdownHelper::convertToHtml($message);
+
+        $form['ai_feedback'] = [
+          '#type' => 'processed_text',
+          '#text' => $message,
+          '#format' => 'html',
+          '#weight' => -90,
+        ];
+      }
+    }
+  }
+
+  // Check permissions.
+  if ($user->hasPermission('bypass ocha ai job tag')) {
+    // Make fields mandatory.
+    $form['field_job_experience']['widget']['#required'] = TRUE;
+    $form['field_career_categories']['widget']['#required'] = TRUE;
+    return;
+  }
+
+  if (!$user->hasPermission('enforce ocha ai job tag')) {
+    // Make fields mandatory.
+    $form['field_job_experience']['widget']['#required'] = TRUE;
+    $form['field_career_categories']['widget']['#required'] = TRUE;
+    return;
+  }
+
+  $sources = array_keys(UserPostingRightsHelper::getUserPostingRights());
+  if (empty($sources)) {
+    // Non-trusted poster, hide fields.
+    $form['field_job_experience']['#access'] = FALSE;
+    $form['field_career_categories']['#access'] = FALSE;
+    $form['field_theme']['#access'] = FALSE;
+    return;
+  }
+
+  // Show and mark fields mandatory for trusted sources.
+  $fields = [
+    'field_job_experience',
+    'field_career_categories',
+    'field_theme',
+  ];
+
+  $conditions = [];
+  foreach ($sources as $source) {
+    $conditions[] = [
+      'value' => $source,
+    ];
+  }
+
+  foreach ($fields as $field) {
+    $form[$field]['#states'] = [
+      'visible' => [
+        'select[name="field_source"]' => $conditions,
+      ],
+      'required' => [
+        'select[name="field_source"]' => $conditions,
+      ],
+    ];
+  }
+}
+
+/**
+ * Implements hook_entity_base_field_info().
+ */
+function reliefweb_job_tagger_entity_base_field_info(EntityTypeInterface $entity_type) {
```

Comment:
@attiks Wouldn't be better to use a bundle field instead of a base field so that it's only added to the jobs and not to all the nodes?

Reply:
True, fixed